src/common/syncjournaldb.cpp

```diff
@@ -935,10 +942,9 @@ bool SyncJournalDb::setFileRecord(const SyncJournalFileRecord &_record)
 
     if (!_avoidReadFromDbOnNextSyncFilter.isEmpty()) {
         // If we are a directory that should not be read from db next time, don't write the etag
-        QByteArray prefix = record._path + "/";
         foreach (const QByteArray &it, _avoidReadFromDbOnNextSyncFilter) {
-            if (it.startsWith(prefix)) {
-                qCInfo(lcDb) << "Filtered writing the etag of" << prefix << "because it is a prefix of" << it;
+            if (it.startsWith(record._path)) {
```

Comment:
This change looks supicious.

So, if `foo/bar/something` is in _avoidReadFromDbOnNextSyncFilter,  that means that we would store the  invalid etag for `foo/b`.

I guess this problem would stay asymptomatic because `foo` normally should have an invalid etag. However, i'd rather not rely on that and have it behave properly as this could otherwise bite us later when the code evolve.

Reply:
Indeed and that was not my intention at all. Instead, I should have made sure `_avoidReadFromDbOnNextSyncFilter` entries always have a trailing slash.